avroregistry/registry.go

```diff
@@ -201,7 +201,7 @@ func (r *Registry) doRequest(req *http.Request, result interface{}) error {
 		if !attempt.More() {
 			return err
 		}
-		if err, ok := err.(*apiError); ok && err.StatusCode/100 != 5 {
+		if err, ok := err.(*apiError); ok && err.StatusCode != http.StatusInternalServerError {
```

Comment:
Any reason to change this? Don't we want to keep retrying proxy errors?

Also, shouldn't we explicitly return `UnavailableError` if we get a 5xx after the max amount of tries?

Reply:
I thought semantically `err.StatusCode/100 != 5` == `err.StatusCode != http.StatusInternalServerError`?

Reply:
> Any reason to change this? Don't we want to keep retrying proxy errors?

I thought semantically err.StatusCode/100 != 5 == err.StatusCode != http.StatusInternalServerError?


> Also, shouldn't we explicitly return `UnavailableError` if we get a 5xx after the max amount of tries?

I have added [it here.](https://github.com/heetch/avro/pull/133/commits/5753d28df13000ea068ce46cc1f17128a0c9a383)

Reply:
You're right. I missed that. Will patch it.